src/components/ModalWithForm.js

```diff
@@ -33,8 +32,14 @@ class ModalWithForm extends Modal {
     return inputValues;
   }
   setEventListeners() {
-    this._modalForm.addEventListener("submit", (event) => {
-      event.preventDefault();
+    this.modalSubmitButton = this._modalForm.querySelector(
+      ".modal__submit-button"
+    );
+
+    this._modalForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.modalSubmitButton.textContent = "Saving...";
+      console.log(this.modalSubmitButton.textContent);
       this.data = this._getInputValues();
       this._handleSubmitForm(this.data);
       this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.